tests/pact/tc.repl

```diff
@@ -335,3 +335,36 @@
  "rtc object lists"
  1
  (rtc-object-list (let ((a {'a:1}) (b {'a: 2})) [a b])))
+
+;; A module with incorrect annotations for a function's return type.
+;; Runtime typechecking for function return types was added in
+;; pact-4.7.
+(module tc-test-function-return-types gov
+  (defcap gov () true)
+  (defun f:string (a: string) (+ 1 2))
+  (defun g:integer (a: integer) a))
+
+;; Functions with incompatible type annotations should not compose.
+(expect-failure
+  "composing type-incompatible, body-incompatible fails evaluation after pact-4.6"
+  (tc-test-function-return-types.g (tc-test-function-return-types.f ""))
```

Comment:
is this necessary? The first test IMO should be simply that `f` throws a runtime error. The fact that it's in an argument position should not matter I think.

Reply:
I think it's good to establish that composition similarly doesn't drop any data. It's good to regress against both, I think, as an inductive base case and subsequent case.

Reply:
It turns out that simply applying the function is enough to trigger typechecking, and the tests fail and pass in the expected way by applying a single function. So I simplified both tests (w/ feature flag and w/out feature flag), to just apply the single function `f` and verify that it fails when expected.

LMK if you think we should bring back `compose` just for an extra level of testing?